client.go

```diff
@@ -43,6 +48,8 @@ type Config struct {
 
 	// PKIClient is the PKI Document data source.
 	PKIClient cpki.Client
+
+	UserKeyDiscovery UserKeyDiscovery
```

Comment:
For me this makes more sense on the Session config. I think the key discovery is dependent of the user.